src/Views/Offline/Detail.js

```diff
@@ -0,0 +1,244 @@
+/**
+ * @class crm.Views.Offline.Detail
+ *
+ *
+ * @extends argos._DetailBase
+ * @requires argos._DetailBase
+ *
+ */
+import declare from 'dojo/_base/declare';
+import _DetailBase from 'argos/_DetailBase';
+import array from 'dojo/_base/array';
+import domConstruct from 'dojo/dom-construct';
+import format from '../../Format';
+import _RelatedWidgetDetailMixin from 'argos/_RelatedViewWidgetDetailMixin';
+import MODEL_TYPES from 'argos/Models/Types';
+import lang from 'dojo/_base/lang';
+import query from 'dojo/query';
+
+
+export default declare('crm.Views.Offline.Detail', [_DetailBase, _RelatedWidgetDetailMixin], {
+  id: 'offline_detail',
+  titleText: 'Offline Detail',
+  offlineText: 'Offline',
+  idProperty: 'id',
+  offlineDoc: null,
+  detailHeaderTemplate: new Simplate([
+    '<div class="detail-header">',
+    '{%: $.value %}',
+    '</div>',
+    '<div class="detail-sub-header">',
+    '{%: $.offlineDate %}',
+    '</div>',
+  ]),
+  relatedTemplate: new Simplate([
+    '<li class="{%= $.cls %}">',
+    '<a data-action="activateRelatedList" data-view="{%= $.view %}" data-name="{%: $.name %}" data-context="{%: $.context %}" {% if ($.disabled) { %}data-disable-action="true"{% } %} class="{% if ($.disabled) { %}disabled{% } %}">',
+    '{% if ($.icon) { %}',
+    '<img src="{%= $.icon %}" alt="icon" class="icon" />',
+    '{% } else if ($.iconClass) { %}',
+    '<div class="{%= $.iconClass %}" alt="icon"></div>',
+    '{% } %}',
+    '<span class="related-item-label">{%: $.label %}</span>',
+    '</a>',
+    '</li>',
+  ]),
+  show: function show(options) {
+    this._initOfflineView(options);
+    this.inherited(arguments);
+  },
+  _initOfflineView: function _initOfflineView(options) {
+    this.offlineContext = {
+      entityName: null,
+      entityId: null,
+      viewId: null,
+      offlineDate: null,
+      source: null,
+    };
+    this.refreshRequired = true;
+    lang.mixin(this.offlineContext, options.offlineContext);
+    this._model = App.ModelManager.getModel(this.offlineContext.entityName, MODEL_TYPES.OFFLINE);
+
+    if (!this.offlineContext.viewId) {
+      this.offlineContext.viewId = (this._model.detailViewId) ? this._model.detailViewId : this._model.entityName.toLowerCase() + '_detail';
+    }
+
+    this._entityView = this.getEntityView();
+  },
+  getEntityView: function getEntityView() {
+    let newView;
+    const newViewId = this.id + '_' + this.offlineContext.viewId;
+    const view = App.getView(this.offlineContext.viewId);
+
+    if (this._entityView) {
```

Comment:
Should this function just set `this._entityView` as well, since all the calls to it just set it?